File: examples/H20/robots/sdk_package/example/h10-w/system/calibrate_mech_zero_pos.cpp
// calibrate_mech_zero_pos.cpp

#include <iostream>
#include "h10-w/h10w_system.h"
#include "common/h10w_def.h"

using namespace humanoid;
using namespace humanoid::h10w;

int main() {
  std::cout << "=== 关节机械零位标定示例 ===" << std::endl;
  H10wSystem system;

  int ret = system.calibrateMechanicalZeroPosition(JointIndexValue::L_ARM_JOINT1);
  std::cout << "标定关节 " << static_cast<uint32_t>(JointIndexValue::L_ARM_JOINT1)
            << (ret == 0 ? " 成功" : " 失败") << std::endl;

  ret = system.saveJointParameters(JointIndexValue::L_ARM_JOINT1);
  std::cout << "保存关节 " << static_cast<uint32_t>(JointIndexValue::L_ARM_JOINT1)
            << (ret == 0 ? " 成功" : " 失败") << std::endl;

  return 0;
}

File: examples/H20/robots/sdk_package/example_py/h10-w/system/calibrate_mech_zero_pos.py
import humanoid_sdk_py
import humanoid_sdk_py.h10w as h10w
import time

h10w_system = h10w.H10wSystem()
time.sleep(1)

print("=== 关节机械零位标定示例 ===")

ret = h10w_system.calibrateMechanicalZeroPosition(h10w.L_ARM_JOINT1)
print("标定关节 ", h10w.L_ARM_JOINT1, ("成功" if ret == 0 else " 失败"))

ret = h10w_system.saveJointParameters(h10w.L_ARM_JOINT1)
print("保存关节 ", h10w.L_ARM_JOINT1, ("成功" if ret == 0 else " 失败"))
